src/codegen/codegen_cpp_visitor.cpp

```diff
@@ -3685,7 +3695,7 @@ void CodegenCppVisitor::print_watch_check() {
     for (int i = 0; i < info.watch_statements.size(); i++) {
         auto statement = info.watch_statements[i];
         auto watch = statement->get_statements().front();
-        auto varname = get_variable_name(fmt::format("watch{}", i + 1));
+        const auto& varname = get_variable_name(fmt::format("watch{}", i + 1));
```

Comment:
maybe worth looking into how the following code generation lines could be done more elegantly - the printing is very fragmented. I have not looked in detail what this does, but just thought I note it down as something to think about.